test/integration/transactions.js

```diff
@@ -281,4 +281,39 @@ describe('bitcoinjs-lib (transactions)', function () {
       assert.strictEqual(keyPair.verify(hash, ss.signature), true)
     })
   })
+
+  it('can verify Transaction (P2SH(P2WPKH)) signatures', function () {
+    const utxos = {
+      'f72d1d83ac40fcedd01415751556a905844ab5f44bbb7728565ebb91b1590109:0': {
+        value: 50000
+      }
+    }
+
+    const txHex = '02000000000101090159b191bb5e562877bb4bf4b54a8405a95615751514d0edfc40ac831d2df7000000001716001435a179e5516947a39ae9c8a25e9fe62c0fc598edffffffff01204e0000000000001976a91431d43308d3c886d53e9ae8a45728370571ff456988ac0247304402206ec41f685b997a51f325b07ee852e82a535f6b52ef54485cc133e05168aa052a022070bafa86108acb51c77b2b259ae8fb7fd1efa10fef804fcfe9b13c2db719acf5012103fb03e9d0a9af86cbed94225dbb8bb70f6b82109bce0a61ddcf41dab6cbb4871100000000'
+    const tx = bitcoin.Transaction.fromHex(txHex)
+
+    tx.ins.forEach(function (input, i) {
+      const txId = Buffer.from(input.hash).reverse().toString('hex')
```

Comment:
This is lame.
Since we have accepted using Object properties... maybe it is time to add `.txId` as a lazy property?

Reply:
maybe...

I agree with the lameness.

Reply:
May be